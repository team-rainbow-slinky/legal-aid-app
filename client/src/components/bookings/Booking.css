.orgBooking {
  text-align: center;
  margin: 2% auto;
  p {
    padding: 2%;
    margin-bottom: 1%;
    border-radius: 4px;
    background: #F8F8F8;
    width: 50%;
    margin: auto;
    span {
      font-weight: bold;
    }
  }
    :hover {
      background-color: #d9ffb3;
      width: 50%;
    }
}

.bookingLink {
    text-decoration: none;
    color: black;
}
