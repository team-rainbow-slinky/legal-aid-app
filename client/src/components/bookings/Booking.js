import React from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';
import styles from './Booking.css';

export default function Booking({ _id, swisId, preferredName }) {
  return (
    <>
      <div className={styles.orgBooking}>
        <h2>{swisId}</h2>
        <p>{preferredName}</p>
      </div>
    </>
  );
}
