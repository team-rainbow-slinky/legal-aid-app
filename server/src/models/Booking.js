import mongoose from 'mongoose';

const bookingSchema = new mongoose.Schema ({
  swisId: {
    type: String,
    required: true
  },
  org: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Org',
    required: true
  },
  preferredName: String,
  gender: String,
  pronouns: String,
  primaryOrgContact: String,
  contacts: String,
  upcomingDates: String,
  meetingHistory: String,
  notes: String,
  mcsoName: String,
  mcsoAge: Number,
  mcsoGender: String,
  mcsoRace: String,
  mcsoHeight: String,
  mcsoWeight: String,
  mcsoHair: String,
  mcsoEyes: String,
  mcsoArrestingAgency: String,
  mcsoBookingDate: String,  
  mcsoAssignedFacility: String,
  mcsoProjectedReleaseDate: String,
  chargesHTML: String
});

export default mongoose.model('Booking', bookingSchema);
