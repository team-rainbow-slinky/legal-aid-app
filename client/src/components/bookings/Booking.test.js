import React from 'react';
import Booking from './Booking';
import { shallow } from 'enzyme';

describe('Booking component', () => {
  it('renders a booking', () => {
    const testBooking = {
      _id: '123',
      preferredName: 'Alex Rankin',
      mcsoName: 'Alexander Rankin',
      mcsoBookingDate: 'December 16, 2018',
      mcsoArrestingAgency: 'Portland Police Bureau'
    };

    const wrapper = shallow(<Booking props={testBooking} />);
    expect(wrapper).toMatchSnapshot();
  });
});
