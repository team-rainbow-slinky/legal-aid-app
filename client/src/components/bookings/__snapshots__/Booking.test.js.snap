// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking component renders a booking 1`] = `
<Link
  className="bookingLink"
  replace={false}
  to="/bookings/undefined"
>
  <div
    className="orgBooking"
  >
    <p>
      <span />
      , booked on 
       by 
    </p>
  </div>
</Link>
`;
